### PR TITLE
Jawn/media renderers

### PR DESCRIPTION
**Pull request by jawndiego**

### CODE REVIEW

File: apps/site/components/client/UploadDialog.tsx [hunk 1]
```diff
@@ -36,30 +37,88 @@ import { FileList } from '@/server'
...
     setFilesToUpload(filesToUpload)
   }, [])
+
   const { getRootProps, getInputProps, isDragActive } = useDropzone({
     onDrop,
     disabled: showFileList,
   })
 
-  const params = useParams()
+  const determineContentType = (file: File): string => {
```
Comment (salieflewis) [inline apps/site/components/client/UploadDialog.tsx:56]: Should this be defined in `lib` instead?

File: apps/site/components/client/renderer/PDF/PDFViewer.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
+'use client'
+import React, { useState } from 'react'
+import { Document, Page, pdfjs } from 'react-pdf'
+
+pdfjs.GlobalWorkerOptions.workerSrc = new URL(
```
Comment (salieflewis) [inline apps/site/components/client/renderer/PDF/PDFViewer.tsx:5]: Why do we need to import this minified js?

File: apps/site/components/client/renderer/PDF/PDFViewer.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+  const onDocumentLoadSuccess = ({ numPages }: { numPages: number }) => {
+    setNumPages(numPages)
+  }
+
+  return (
+    <div className="flex flex-col items-center my-4">
+      <div
+        className="pdf-container overflow-auto mb-4"
+        style={{ height: '70vh' }}
```
Comment (salieflewis) [inline apps/site/components/client/renderer/PDF/PDFViewer.tsx:26]: Can we achieve this styling with tailwind?

File: apps/site/components/client/renderer/PDF/index.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+'use client'
```
Comment (salieflewis) [inline apps/site/components/client/renderer/PDF/index.ts:1]: Think we should make a decision about whether or not we're adding the directives using barrel files or directly in the files themselves.

File: apps/site/components/client/renderer/glb/Model.tsx [hunk 5]
```diff
@@ -1,57 +1,50 @@
...
   return (
+    // we can create a fall back loading component if we wish
     <Canvas>
-      <ambientLight intensity={1} />
-      <directionalLight position={[0, 5, 5]} intensity={2} />
-      <directionalLight position={[5, 5, 5]} intensity={3} />
-
-      <primitive object={gltf.scene} dispose={null} />
-      <OrbitControls />
+      <Suspense fallback={null}>
```
Comment (salieflewis) [inline apps/site/components/client/renderer/glb/Model.tsx:37]: Can we decide upon a fallback here? Or just remove it if not?

File: apps/site/components/client/renderer/glb/Model.tsx [hunk 6]
```diff
@@ -1,57 +1,50 @@
...
-//     //   ar: true,
-//     //   'ar-modes': "webxr scene-viewer quick-look",
-//     //   'environment-image': "neutral"
-//     // } as Partial<ModelViewerElement>)
-//   );
-// }
-
-// export default GLTFViewer;
\ No newline at end of file
+export default ModelViewer
```
Comment (salieflewis) [inline apps/site/components/client/renderer/glb/Model.tsx:50]: Can we avoid using default exports? Just based on the fact that we're not using this pattern elsewhere.

File: apps/site/lib/isContent.ts [hunk 7]
```diff
@@ -16,3 +32,21 @@ export function isGLB(file: File) {
   const extension = file.name.split('.').pop()?.toLowerCase()
   return extension === 'glb' || extension === 'gltf'
 }
+
+export function isPDF({ mimeType }: { mimeType: string }) {
```
Comment (salieflewis) [inline apps/site/lib/isContent.ts:36]: Thoughts about instances of 'PDF' just reading 'Pdf', as to not make exceptions with existing things that are typically written in all uppercase?

i.e. CID, URL, URI

**Review by salieflewis — APPROVED** [review]
Left, mostly nits...

cc: @jawndiego

File: apps/site/components/client/renderer/PDF/PDFViewer.tsx (hunk 2 above)
Comment (jawndiego) [reply apps/site/components/client/renderer/PDF/PDFViewer.tsx:5]: its something that helps render the info i think a lot of the actual functionality is aided by that worker. pdf js doesnt work without it unfortunately.

File: apps/site/components/client/renderer/PDF/PDFViewer.tsx (hunk 3 above)
Comment (jawndiego) [reply apps/site/components/client/renderer/PDF/PDFViewer.tsx:26]: pain i thought this was tail wind.

File: apps/site/components/client/renderer/glb/Model.tsx (hunk 6 above)
Comment (jawndiego) [reply apps/site/components/client/renderer/glb/Model.tsx:50]: i tried to do it where we do 'export const modelviewer' for example and then i change the dynamic loading but i encounter several bugs when i make that change. i think its worth creating an issue to revert back to using the pattern we've already set.

**Pull request merged**